index.ts

```diff
@@ -0,0 +1,186 @@
+import psi from 'psi';
+// https://stackoverflow.com/a/62499498/9931154
+import { createRequire } from 'module';
+import { type Options } from './options.js';
+const metaRequire = createRequire(import.meta.url);
+const lighthouseCli = metaRequire.resolve('lighthouse/lighthouse-cli');
+import { v4 as uuid } from 'uuid';
+import fs from 'fs';
+import chalk from 'chalk';
+
+const NUM_RUNS = 5;
+const platforms = ['mobile', 'desktop'] as const;
+
+const colorMap = {
+    bg: {
+        red: chalk.bgRed,
+        redBright: chalk.bgRedBright,
+        yellow: chalk.bgYellow,
+        yellowBright: chalk.bgYellowBright,
+        green: chalk.bgGreen,
+        greenBright: chalk.bgGreenBright,
+    },
+    fg: {
+        red: chalk.red,
+        redBright: chalk.redBright,
+        yellow: chalk.yellow,
+        yellowBright: chalk.yellowBright,
+        green: chalk.green,
+        greenBright: chalk.greenBright,
+    },
+};
+
+const colorScore = (score: number, bg = false) => {
+    const map = bg ? colorMap.bg : colorMap.fg;
+    switch (true) {
+        case score < 30:
+            return map.red(score);
+        case score < 50:
+            return map.redBright(score);
+        case score < 70:
+            return map.yellow(score);
+        case score < 90:
+            return map.yellowBright(score);
+        case score < 95:
+            return map.green(score);
+        case score <= 100:
+            return map.greenBright(score);
+    }
+};
+const processScore = (score: object) => Math.round(parseFloat(score.toString()) * 100);
+
+const colorPlatform = (platform: string) => {
+    switch (platform) {
+        case 'mobile':
+            return chalk.magenta(platform);
+        case 'desktop':
+            return chalk.cyan(platform);
+        default:
+            return chalk.gray(platform);
+    }
+};
+
+const localRun = (urlWithRun: string) => {
+    // TODO: allow local to simulate mobile. Will only simulate desktop.
+    const { status = -1, stdout } = spawnSync('node', [
+        lighthouseCli,
+        '--config-path=./node_modules/lighthouse/lighthouse-core/config/lr-desktop-config.js',
+        urlWithRun,
+        '--chromeFlags="--headless"',
+        '--output=json',
+        '--throttling.rttMs=40',
+        '--throttling.throughputKbps=10240',
+        // https://lighthouse-cpu-throttling-calculator.vercel.app/
+        '--throttling.cpuSlowdownMultiplier=6',
+    ]);
+    if (status !== 0) {
+        throw new Error('Lighthouse failed, skipping run...');
+    }
+    return JSON.parse(stdout.toString());
+};
+
+const psiRun = async (
+    urlWithRun: string,
+    key: string,
+    platform: (typeof platforms)[number]
+): Promise<psi.LighthouseResult> => {
+    const { data } = await psi(urlWithRun, {
+        key,
+        strategy: platform,
+        // @ts-ignore -- this is necessary for the current setup to work but does not exist in the types
```

Comment:
Because we expect an error, could we switch this directive to make that clearer? Thanks!

```suggestion
        // @ts-expect-error -- this is necessary for the current setup to work but does not exist in the types
```

Reply:
Yeah, I saw this `@ts-expect-error` for the first time yesterday as well, will do!